### PR TITLE
update get_lowest_taxonomy() and remove horizontal whitelines from heatmap

### PR DESCRIPTION
**Pull request by RNAer**

### CODE REVIEW

**Comment by amnona** [comment]
travis error:
flake8 calour setup.py
calour/tests/test_amplicon_experiment.py:111:14: E225 missing whitespace around operator
make: *** [pep8] Error 1
The command "WITH_COVERAGE=TRUE make all" exited with 2.

File: calour/heatmap/heatmap.py [hunk 1]
```diff
@@ -89,46 +89,53 @@ def _truncate_middle(x, length=16):
...
     '''Plot tick and ticklabels of x or y axis.
 
     Parameters
     ----------
     ax : matplotlib.axes.Axes
         The axes to plot into
-    Parameters mean axes, x or y axis, list of tick labels, max number
+    Parameters: axes, x or y axis, list of tick labels, max number
     of ticks, the upper bound of xlim/ylim, dict passing as text
     property, the length of each tick label.
```
Comment (amnona) [inline calour/heatmap/heatmap.py]: can you add a description of transition parameter?

File: calour/heatmap/heatmap.py [hunk 2]
```diff
@@ -319,7 +326,7 @@ def heatmap(exp: Experiment, sample_field=None, feature_field=None,
         except KeyError:
             raise ValueError('Feature field %r not in feature metadata' % feature_field)
         # numcols instead of numrows because it is transposed
-        _set_axis_ticks(ax, 'y', ticklabels, yticks_max, numcols, yticklabel_kwargs, yticklabel_len)
+        _set_axis_ticks(ax, 'y', ticklabels, yticks_max, numcols, yticklabel_kwargs, yticklabel_len, transition=False)
```
Comment (amnona) [inline calour/heatmap/heatmap.py]: What does it mean we send transition=False? Is there a way to override?

File: calour/heatmap/heatmap.py (hunk 1 above)
Comment (amnona) [inline calour/heatmap/heatmap.py]: can you add a description of transition parameter?

File: calour/heatmap/heatmap.py [hunk 3]
```diff
@@ -490,13 +497,13 @@ def plot(exp: Experiment, title=None,
...
+        The thickness of the each bar along x axis or y axis. The default thickness usually looks good enough.
     barx_colors : dict, matplotlib.colors.ListedColormap, optional
     bary_colors : dict, matplotlib.colors.ListedColormap, optional
         The colors for each unique values in the column of sample/feature metadata
-    barx_label : bool, optional
-    bary_label : bool, optional
-        whether to show the labels on the bars.
+    barx_label : bool or list of bool, optional
+    bary_label : bool or list of bool, optional
+        whether to show the labels on the bars along x axis or y axis.
```
Comment (amnona) [inline calour/heatmap/heatmap.py]: What does list of bool do? Is it one per bar?

File: calour/heatmap/heatmap.py (hunk 2 above)
Comment (RNAer) [reply calour/heatmap/heatmap.py]: not in this function's API. I feel like in our use cases, there is no need to collapse/group y tick label and white horizontal line is distracting. so I don't want to add another function parameter here.

File: calour/heatmap/heatmap.py (hunk 3 above)
Comment (RNAer) [reply calour/heatmap/heatmap.py]: correct

File: calour/heatmap/heatmap.py (hunk 1 above)
Comment (RNAer) [reply calour/heatmap/heatmap.py]: added

File: calour/heatmap/heatmap.py (hunk 2 above)
Comment (amnona) [reply calour/heatmap/heatmap.py]: i used the y axis bars a few times when grouping by taxonomy and labeling with taxonomy names.
I think better to keep an option (and it's also symmetric with the x-axis, so does not complicate things...)?
can discuss tomorrow

**Pull request merged**